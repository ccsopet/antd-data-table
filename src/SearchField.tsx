import * as React from 'react'
import {
  Button,
  Form,
  Table,
  Row,
  Col,
  Icon,
  Input
} from 'antd'
import { WrappedFormUtils } from 'antd/lib/form/Form'
import { IDataTableProps } from './'
const FormItem = Form.Item

const comesWithRenderer = {
  input: require('./renderer/input').default
}

/** Your component's props */
export interface ISearchFieldProps extends IDataTableProps {
  /** antd form instance */
  form?: WrappedFormUtils,
  /** patch response result handler */
  applyData: (data: any[]) => void
}

/** Your component's state */
export interface ISearchFieldState {
  expand: boolean,
  loading: boolean
}

/** Your component */
export class SearchField extends React.Component<ISearchFieldProps, ISearchFieldState> {

  state = {
    expand: false,
    loading: false
  }

  private shouldHandleCollapse = this.props.maxVisibleFieldCount && this.props.searchFields.length > this.props.maxVisibleFieldCount

  toggleExpand = () => {
    const { expand } = this.state
    this.setState({ expand: !expand })
  }

  getFields = () => {
    const { form, maxVisibleFieldCount, searchFields } = this.props
    if (!form) { return false }
    const { getFieldDecorator } = form
    const formItemLayout = {
      labelCol: { span: 5 },
      wrapperCol: { span: 19 }
    }
    const count = this.state.expand ? searchFields.length : maxVisibleFieldCount || searchFields.length
    return this.props.searchFields.map((searchField, i) => {
      const renderComponent = () => {
        if (searchField.renderer) {
          // 自定义 renderer
          return searchField.renderer(searchField.payload)
        } else {
          // 自带 renderer
          if (searchField.type) {
            if (comesWithRenderer[searchField.type]) {
              return comesWithRenderer[searchField.type](searchField.payload)
            } else {
              console.warn('Unknown renderer:', searchField.type)
              return false
            }
          } else {
            // 既没有 type 又没有 renderer
            console.warn('Renderer or Type should exist in search field')
            return false
          }
        }
      }
      return (
        <Col span={8} key={i} style={this.shouldHandleCollapse ? { display: i < count ? 'block' : 'none' } : { display: 'block' }}>
          <FormItem {...formItemLayout} label={searchField.label}>
            {getFieldDecorator(searchField.name, { rules: searchField.validationRule })(
              renderComponent()
            )}
          </FormItem>
        </Col>
      )
    })
  }

  clearField = () => {
    const { form } = this.props
    if (!form) { return false }

    form.resetFields()
  }

  onSearch = () => {
    const { form, onError, onSearch, onValidateFailed, applyData } = this.props
    if (!form) { return false }
    const { validateFields } = form

    validateFields(async (err, values) => {
      if (err) {
        onValidateFailed && onValidateFailed(err)
        return
      }

      try {
        this.setState({
          loading: true
        })
        const res = await this.props.onSearch({
          page: 1,
          pageSize: 10,
          values
        })
        applyData(res.data)
      } catch (e) {
        onError && onError(e)
      } finally {
        this.setState({
          loading: false
        })
      }
    })
  }

  render () {
    return (
      <Form
        className='ant-advanced-search-form'
      >
        <Row gutter={40}>{this.getFields()}</Row>
        <Row>
          <Col span={24} style={{ textAlign: 'right' }}>
            <Button type='primary' onClick={this.onSearch} loading={this.state.loading}>Search</Button>
            <Button style={{ marginLeft: 8 }} onClick={this.clearField}>
              Clear
            </Button>
            {this.shouldHandleCollapse && (
              <a style={{ marginLeft: 8, fontSize: 12 }} onClick={this.toggleExpand}>
                Collapse <Icon type={this.state.expand ? 'up' : 'down'} />
              </a>
            )}
          </Col>
        </Row>
      </Form>
    )
  }
}

/** Export as default */
export default Form.create<ISearchFieldProps>()(SearchField as any)
