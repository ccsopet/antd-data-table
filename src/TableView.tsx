import * as React from 'react'
import {
  Table
} from 'antd'
import { TableColumnConfig } from 'antd/lib/table/Table'
import { PaginationProps } from 'antd/lib/pagination/Pagination'
import { IDataTableProps, SearchInfo, SearchFunc } from './'

export interface ITableViewProps extends IDataTableProps {
  data: any[],
  columns: TableColumnConfig<any>[],
  pagination: PaginationProps,
  loading: boolean,
  title?: (currentPageData) => React.ReactNode,
  onTableChange: (pagination: PaginationProps | boolean, filters: string[], sorter: Object) => any,
  fetch: SearchFunc
}

export interface ITableViewState {
}

class TableView extends React.Component<ITableViewProps, ITableViewState> {

  render () {
    return (
      <Table
        title={this.props.title}
        loading={this.props.loading}
        columns={this.props.columns}
        dataSource={this.props.data}
        onChange={this.props.onTableChange}
        pagination={this.props.pagination}
      />
    )
  }
}

export default TableView
