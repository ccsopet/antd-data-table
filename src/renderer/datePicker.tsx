import * as React from 'react'

import {
  DatePicker
} from 'antd'
import { SearchFieldPayload } from '../'

export default (payload?: SearchFieldPayload) => {
  return (
    <DatePicker {...payload && payload.props} />
  )
}
