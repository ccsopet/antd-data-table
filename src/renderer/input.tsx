import * as React from 'react'
import {
  Input
} from 'antd'

export default payload => {
  return (
    <Input {...this.payload} />
  )
}
