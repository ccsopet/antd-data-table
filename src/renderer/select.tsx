import * as React from 'react'
import {
  Select
} from 'antd'
import { SearchFieldPayload } from '../'

type SelectOption = {
  key: string,
  label: string,
  value: string
}

export default (payload?: SearchFieldPayload) => {
  if (!payload || !payload.options) {
    console.warn('select renderere expected `options`')
    return null
  }
  const options = payload.options as SelectOption[]
  return (
    <Select {...payload && payload.props}>
      {options.map(option => {
        return (
          <Select.Option key={option.key} value={option.value}>
            {option.label}
          </Select.Option>
        )
      })}
    </Select>
  )
}
