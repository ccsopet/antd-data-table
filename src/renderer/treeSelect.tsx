import * as React from 'react'

import {
  TreeSelect
} from 'antd'
import { SearchFieldPayload } from '../'

export default (payload?: SearchFieldPayload) => {
  return (
    <TreeSelect {...payload && payload.props} />
  )
}
