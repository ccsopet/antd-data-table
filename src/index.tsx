import * as React from 'react'
import {
  Button,
  Form,
  Table,
  Row,
  Col,
  Icon,
  Dropdown,
  Card,
  Checkbox,
  Menu
} from 'antd'
import * as update from 'immutability-helper'
import { TableColumnConfig } from 'antd/lib/table/Table'
import { PaginationProps } from 'antd/lib/pagination/Pagination'

import { ValidationRule } from 'antd/lib/form/Form'
import SearchField, { ISearchFieldProps } from './SearchField'

export type ValidateError = {
  [fieldName: string]: {
    errors: {
      message: string,
      field: string
    }[]
  }
}

export type SearchInfo = {
  values: any,
  page: number,
  pageSize: number
}

export type SearchFunc<T = void> = (page: number, values?: object, clearPagination?: boolean) => Promise<T>

export type SearchResponse<T> = {
  dataSource: T[],
  total: number
}

export type FieldRenderer = (payload?: object) => React.ReactNode

export type RendererType = 'input' | 'select'

export type SearchField = {
  label: string,
  name: string,
  type?: RendererType,
  renderer?: FieldRenderer,
  validationRule?: ValidationRule[],
  payload?: SearchFieldPayload
}

export type SearchFieldPayload = {
  /** props that pass to the main component */
  props?: object,
  [key: string]: any
}

export type RowAction = {
  label: string,
  children?: RowAction[],
  action?: (record) => void
}

export type Plugin = {
  colSpan?: number,
  renderer: (selectedRowKeys: string[], selectedRows: any[], clearSelectionCallback: () => void) => React.ReactNode
}

/** Your component's props */
export interface IDataTableProps {
  initialColumns: TableColumnConfig<any>[],
  searchFields: SearchField[],
  rowActions?: RowAction[],
  plugins?: Plugin[],
  rowKey: (record: any) => string,
  title?: React.ReactNode,
  searchBtnText?: string,
  clearBtnText?: string,
  listSelectionBtnText?: string,
  /** 最大的表单项显示数，当表单项超过此数值时，会自动出现 collapse 按钮 */
  maxVisibleFieldCount?: number,
  pageSize?: number,
  /** handle form validate error */
  onValidateFailed?: (err: ValidateError) => void,
  /** 执行 search 动作，返回一个 AxiosPromis */
  onSearch<T> (info: SearchInfo): Promise<SearchResponse<T>>,
  /** reject handler */
  onError? (err): void
}

/** Your component's state */
export interface IDataTableState {
  columns: TableColumnConfig<any>[],
  data: any[],
  page: number,
  currentValues: object,
  pagination: PaginationProps,
  tableLoading: boolean,
  searchButtonLoading: boolean,
  selectedRowKeys: string[],
  selectedRows: any[]
}

const renderActions = (actions: RowAction[], record) => {
  return (
    <span>
      {actions.map((action, i) => {
        if (action.children) {
          const menu = (
            <Menu>
              {action.children.map(child => {
                const onClick = () => {
                  child.action && child.action(record)
                }
                return (
                  <Menu.Item>
                    <a onClick={onClick}>{child.label}</a>
                  </Menu.Item>
                )
              })}
            </Menu>
          )
          return (
            <Dropdown overlay={menu}>
              <a className='ant-dropdown-link'>
                {action.label} <Icon type='down' />
              </a>
            </Dropdown>
          )
        } else {
          const onClick = () => {
            action.action && action.action(record)
          }
          return [
            <a onClick={onClick}>{action.label}</a>,
            i === 0 && <span className='ant-divider' />
          ]
        }
      })}
    </span>
  )
}

/** Your component */
export class DataTable extends React.Component<IDataTableProps, IDataTableState> {

  static defaultProps = {
    pageSize: 10,
    searchBtnText: 'Search',
    clearBtnText: 'Clear',
    listSelectionBtnText: 'List selection'
  }

  actionsColumn = this.props.rowActions && { key: 'actions', title: 'Actions', render: (record) => { return renderActions(this.props.rowActions as RowAction[], record) } } as TableColumnConfig<any>

  initialColumns = this.actionsColumn ? [...this.props.initialColumns, this.actionsColumn] : this.props.initialColumns

  state = {
    columns: [] = this.initialColumns,
    data: [],
    page: 1,
    pagination: {} as PaginationProps,
    currentValues: {},
    tableLoading: false,
    searchButtonLoading: false,
    selectedRows: [],
    selectedRowKeys: []
  }

  filterPannel = (<Card bodyStyle={{ padding: '1em', width: '12em' }}>
    {this.initialColumns.map(column => {
      const isSelected = this.state.columns.find(c => c.key === column.key) !== undefined
      const onChange = (e) => {
        if (e.target.checked) {
          this.showColumn(column.key)
        } else {
          this.hideColumn(column.key)
        }
      }
      return (
        <p key={column.key} style={{ marginTop: '.5em', marginBottom: '.5em' }}>
          <Checkbox defaultChecked={isSelected} onChange={onChange}>{column.title as any}</Checkbox>
        </p>
      )
    })}
  </Card>)

  startTableLoading = () => {
    this.setState({ tableLoading: true })
  }

  stopTableLoading = () => {
    this.setState({ tableLoading: false })
  }

  startSearchButtonLoading = () => {
    this.setState({ searchButtonLoading: true })
  }

  stopSearchButtonLoading = () => {
    this.setState({ searchButtonLoading: false })
  }

  tableTitle = (currentPageData) => {
    return (
      <Row type='flex'>
        <Col span={12}>
          {this.props.title}
        </Col>
        <Col span={12}>
          <Row type='flex' justify='end'>
            <Col>
              <Dropdown overlay={this.filterPannel} trigger={['click']}>
                <Button size='small'>{this.props.listSelectionBtnText}</Button>
              </Dropdown>
            </Col>
          </Row>
        </Col>
      </Row>
    )
  }

  applyData = (data: any[]) => {
    this.setState({ data })
  }

  applyValues = (values, cb) => {
    this.setState({ currentValues: values }, cb)
  }

  clearPagination = () => {
    const pager = { ...this.state.pagination }
    pager.current = 1
    this.setState({ pagination: pager })
  }

  handleChange = async (pagination: PaginationProps) => {
    const { onError } = this.props
    const pager = { ...this.state.pagination }
    pager.current = pagination.current
    this.setState({ pagination: pager })
    this.fetch(pager.current || 1)
  }

  fetch: SearchFunc = async (page: number, values: object = this.state.currentValues, clearPagination: boolean = false) => {
    const { onError } = this.props
    this.applyValues(values, async () => {
      try {
        // 这里先简单认为 clearPagination 为 true 就是从 Search button 触发的 fetch
        clearPagination && this.startSearchButtonLoading()
        this.startTableLoading()
        const pager = { ...this.state.pagination }
        const response = await this.props.onSearch({
          page: page,
          // pageSize 有 default
          pageSize: this.props.pageSize as number,
          values: this.state.currentValues
        })
        pager.total = response.total
        this.setState({
          pagination: pager
        })
        this.applyData(response.dataSource)
        clearPagination && this.clearPagination()
      } catch (e) {
        onError && onError(e)
      } finally {
        clearPagination && this.stopSearchButtonLoading()
        this.stopTableLoading()
      }
    })
  }

  hideColumn = (key?: string) => {
    this.state.columns.forEach((column, i) => {
      if (column.key === key) {
        this.setState({
          columns: update(this.state.columns, { $splice: [[i, 1]] })
        })
      }
    })
  }

  clearSelection = () => {
    this.setState({
      selectedRows: [],
      selectedRowKeys: []
    })
  }

  showColumn = (key?: string) => {
    this.initialColumns.forEach((column, i) => {
      if (column.key === key) {
        this.setState({
          columns: update(this.state.columns, { $splice: [[i, 0, column]] })
        })
      }
    })
  }

  render () {
    const rowSelection = {
      selectedRowKeys: this.state.selectedRowKeys,
      onChange: (selectedRowKeys, selectedRows) => {
        this.setState({
          selectedRowKeys, selectedRows
        })
      }
    }

    return (
      <div>
        <div>
          <SearchField {...this.props} fetch={this.fetch} btnLoading={this.state.searchButtonLoading} />
        </div>
        <div>
          {this.props.plugins && <Row className='operationpannel' gutter={16} type='flex' style={{ paddingBottom: '1em' }}>
            {(this.props.plugins).map(plugin => {
              return (
                <Col span={plugin.colSpan}>
                  {plugin.renderer(this.state.selectedRowKeys, this.state.selectedRows, this.clearSelection)}
                </Col>
              )
            })}
          </Row>}
          <Row>
            <Table
              bordered
              size='middle'
              title={this.tableTitle}
              rowSelection={rowSelection}
              rowKey={this.props.rowKey}
              loading={this.state.tableLoading}
              columns={this.state.columns}
              dataSource={this.state.data}
              onChange={this.handleChange}
              pagination={this.state.pagination}
            />
          </Row>
        </div>
      </div>
    )
  }
}

/** Export as default */
export default DataTable
