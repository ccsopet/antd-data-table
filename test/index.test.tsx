import * as React from 'react'
import renderer from 'react-test-renderer'
import { shallow, render } from 'enzyme'
import { DataTable, SearchField, SearchInfo } from '../src'
import { storiesOf } from '@storybook/react'

import axios from 'axios'

import { TableColumnConfig } from 'antd/lib/table/Table'

/** Import component */

const onSearch = (info: SearchInfo) => {
  return axios.get('http://jsonplaceholder.typicode.com/posts', {
    params: {
      _page: info.page,
      _limit: info.pageSize,
      ...info.values
    }
  })
}

const columns: TableColumnConfig<any>[] = [
  {
    key: 'id',
    title: 'ID',
    dataIndex: 'id'
  }, {
    key: 'title',
    title: 'Title',
    dataIndex: 'title'
  }
]

const searchFields: SearchField[] = [
  {
    label: 'ID',
    name: 'id',
    type: 'input',
    payload: {
      props: {
        placeholder: 'placeholder'
      }
    }
  },
  {
    label: 'Select',
    name: 'select',
    type: 'select',
    payload: {
      props: {
        allowClear: true
      },
      options: [
        { key: '1', label: 'one', value: '1' },
        { key: '2', label: 'two', value: '2' },
        { key: '3', label: 'three', value: '3' }
      ]
    }
  },
  {
    label: 'Multi Select',
    name: 'multi-select',
    type: 'select',
    payload: {
      props: {
        mode: 'multiple'
      },
      options: [
        { key: '1', label: 'one', value: '1' },
        { key: '2', label: 'two', value: '2' },
        { key: '3', label: 'three', value: '3' }
      ]
    }
  }
]

describe('DataTable', () => {

  /** Snapshot testing */
  it('should render correctly', () => {
    let wrapper = shallow(<DataTable
      searchFields={searchFields}
      initialColumns={columns}
      onSearch={onSearch}
      pageSize={10}
    />)

    expect(wrapper).toMatchSnapshot()
  })

  /** Enzyme shallow testing */
  it('should have one div', () => {
    // let wrapper = shallow(<DataTable />)
    // expect(wrapper.find('div').length).toEqual(1)
  })
})
