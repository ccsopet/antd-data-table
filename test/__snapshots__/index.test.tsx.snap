// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable should render correctly 1`] = `
<div>
  <div>
    <form
      className="ant-form ant-form-horizontal ant-advanced-search-form"
      onSubmit={[Function]}
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -20,
            "marginRight": -20,
          }
        }
      >
        <div
          className="ant-col-6"
          style={
            Object {
              "display": "block",
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <div
            className="ant-row ant-form-item"
            style={undefined}
          >
            <div
              className="ant-col-8 ant-form-item-label"
            >
              <label
                className=""
                htmlFor="id"
                onClick={[Function]}
                title="ID"
              >
                ID
              </label>
            </div>
            <div
              className="ant-col-16 ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control "
              >
                <input
                  className="ant-input ant-input-lg"
                  data-__meta={
                    Object {
                      "initialValue": undefined,
                      "leadingName": "id",
                      "name": "id",
                      "originalProps": Object {
                        "disabled": false,
                        "placeholder": "placeholder",
                        "prefixCls": "ant-input",
                        "type": "text",
                      },
                      "ref": null,
                      "rules": undefined,
                      "trigger": "onChange",
                      "validate": Array [],
                      "valuePropName": "value",
                    }
                  }
                  disabled={false}
                  id="id"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="placeholder"
                  size="large"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col-6"
          style={
            Object {
              "display": "block",
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <div
            className="ant-row ant-form-item"
            style={undefined}
          >
            <div
              className="ant-col-8 ant-form-item-label"
            >
              <label
                className=""
                htmlFor="select"
                onClick={[Function]}
                title="Select"
              >
                Select
              </label>
            </div>
            <div
              className="ant-col-16 ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control has-success"
              >
                <div
                  className="ant-select-lg ant-select ant-select-enabled ant-select-allow-clear"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={undefined}
                  onMouseLeave={undefined}
                  onTouchStart={[Function]}
                  style={undefined}
                >
                  <div
                    aria-autocomplete="list"
                    aria-expanded={false}
                    aria-haspopup="true"
                    className="ant-select-selection
                              ant-select-selection--single"
                    onKeyDown={[Function]}
                    role="combobox"
                    tabIndex={0}
                  >
                    <div
                      className="ant-select-selection__rendered"
                    >
                      <div
                        className="ant-select-selection-selected-value"
                        style={
                          Object {
                            "display": "block",
                            "opacity": 1,
                          }
                        }
                        title="one"
                      >
                        one
                      </div>
                    </div>
                    <span
                      className="ant-select-selection__clear"
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      style={
                        Object {
                          "WebkitUserSelect": "none",
                          "display": "block",
                          "userSelect": "none",
                        }
                      }
                      unselectable="unselectable"
                    />
                    <span
                      className="ant-select-arrow"
                      onClick={[Function]}
                      style={
                        Object {
                          "WebkitUserSelect": "none",
                          "userSelect": "none",
                        }
                      }
                      unselectable="unselectable"
                    >
                      <b />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col-6"
          style={
            Object {
              "display": "block",
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <div
            className="ant-row ant-form-item"
            style={undefined}
          >
            <div
              className="ant-col-8 ant-form-item-label"
            >
              <label
                className=""
                htmlFor="multi-select"
                onClick={[Function]}
                title="Multi Select"
              >
                Multi Select
              </label>
            </div>
            <div
              className="ant-col-16 ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control "
              >
                <div
                  className="ant-select-lg ant-select ant-select-enabled"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={undefined}
                  onMouseLeave={undefined}
                  onTouchStart={[Function]}
                  style={undefined}
                >
                  <div
                    aria-autocomplete="list"
                    aria-expanded={false}
                    aria-haspopup="true"
                    className="ant-select-selection
                              ant-select-selection--multiple"
                    role="combobox"
                  >
                    <div
                      className="ant-select-selection__rendered"
                    >
                      <ul
                        className={undefined}
                        style={undefined}
                      >
                        <li
                          className="ant-select-search ant-select-search--inline"
                        >
                          <div
                            className="ant-select-search__field__wrap"
                          >
                            <input
                              autoComplete="off"
                              className="ant-select-search__field"
                              disabled={undefined}
                              id="multi-select"
                              onChange={[Function]}
                              onKeyDown={[Function]}
                              value=""
                            />
                            <span
                              className="ant-select-search__field__mirror"
                            >
                              
                              Â 
                            </span>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col-6"
          style={
            Object {
              "display": "block",
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <div
            className="ant-row ant-form-item"
            style={undefined}
          >
            <div
              className="ant-col-8 ant-form-item-label"
            >
              <label
                className=""
                htmlFor="datePicker"
                onClick={[Function]}
                title="Date Picker"
              >
                Date Picker
              </label>
            </div>
            <div
              className="ant-col-16 ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control "
              >
                <span
                  className="ant-calendar-picker"
                  style={Object {}}
                >
                  <div
                    onBlur={undefined}
                    onClick={[Function]}
                    onFocus={undefined}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={undefined}
                    onMouseLeave={undefined}
                    onTouchStart={[Function]}
                  >
                    <input
                      className="ant-calendar-picker-input ant-input ant-input-lg"
                      disabled={undefined}
                      placeholder="è¯·éæ©æ¥æ"
                      readOnly={true}
                      value=""
                    />
                    <span
                      className="ant-calendar-picker-icon"
                    />
                  </div>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col-6"
          style={
            Object {
              "display": "block",
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <div
            className="ant-row ant-form-item"
            style={undefined}
          >
            <div
              className="ant-col-8 ant-form-item-label"
            >
              <label
                className=""
                htmlFor="treeselect"
                onClick={[Function]}
                title="Tree Select"
              >
                Tree Select
              </label>
            </div>
            <div
              className="ant-col-16 ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control "
              >
                <span
                  className="ant-select-lg ant-select ant-select-enabled"
                  onBlur={undefined}
                  onClick={[Function]}
                  onFocus={undefined}
                  onMouseDown={[Function]}
                  onMouseEnter={undefined}
                  onMouseLeave={undefined}
                  onTouchStart={[Function]}
                  style={undefined}
                >
                  <span
                    aria-autocomplete="list"
                    aria-expanded={undefined}
                    aria-haspopup="true"
                    className="ant-select-selection
                              ant-select-selection--single"
                    onKeyDown={[Function]}
                    role="combobox"
                    tabIndex={0}
                  >
                    <span
                      className="ant-select-selection__rendered"
                    >
                      <span
                        className="ant-select-selection__placeholder"
                      >
                        
                      </span>
                    </span>
                    <span
                      className="ant-select-arrow"
                      style={
                        Object {
                          "outline": "none",
                        }
                      }
                    >
                      <b />
                    </span>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col-6"
          style={
            Object {
              "display": "block",
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <div
            className="ant-row ant-form-item"
            style={undefined}
          >
            <div
              className="ant-col-8 ant-form-item-label"
            >
              <label
                className=""
                htmlFor="foo"
                onClick={[Function]}
                title="Foo"
              >
                Foo
              </label>
            </div>
            <div
              className="ant-col-16 ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control "
              >
                <input
                  className="ant-input ant-input-lg"
                  data-__meta={
                    Object {
                      "initialValue": undefined,
                      "leadingName": "foo",
                      "name": "foo",
                      "originalProps": Object {
                        "disabled": false,
                        "prefixCls": "ant-input",
                        "type": "text",
                      },
                      "ref": null,
                      "rules": undefined,
                      "trigger": "onChange",
                      "validate": Array [],
                      "valuePropName": "value",
                    }
                  }
                  disabled={false}
                  id="foo"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  size="large"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col-6"
          style={
            Object {
              "display": "block",
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <div
            className="ant-row ant-form-item"
            style={undefined}
          >
            <div
              className="ant-col-8 ant-form-item-label"
            >
              <label
                className=""
                htmlFor="bar"
                onClick={[Function]}
                title="Bar"
              >
                Bar
              </label>
            </div>
            <div
              className="ant-col-16 ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control "
              >
                <input
                  className="ant-input ant-input-lg"
                  data-__meta={
                    Object {
                      "initialValue": undefined,
                      "leadingName": "bar",
                      "name": "bar",
                      "originalProps": Object {
                        "disabled": false,
                        "prefixCls": "ant-input",
                        "type": "text",
                      },
                      "ref": null,
                      "rules": undefined,
                      "trigger": "onChange",
                      "validate": Array [],
                      "valuePropName": "value",
                    }
                  }
                  disabled={false}
                  id="bar"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  size="large"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row"
        style={undefined}
      >
        <div
          className="ant-col-24"
          style={
            Object {
              "textAlign": "right",
            }
          }
        >
          <button
            className="ant-btn ant-btn-primary"
            onClick={[Function]}
            type="button"
          >
            <span>
              Search
            </span>
          </button>
          <button
            className="ant-btn"
            onClick={[Function]}
            style={
              Object {
                "marginLeft": 8,
              }
            }
            type="button"
          >
            <span>
              Clear
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
  <div>
    <div
      className="ant-row"
      style={undefined}
    >
      <div
        className="ant-table-wrapper"
        style={undefined}
      >
        <div
          className="ant-spin-nested-loading"
          style={null}
        >
          <div
            className="ant-spin-container"
          >
            <div
              className="ant-table ant-table-middle ant-table-bordered ant-table-empty ant-table-scroll-position-left"
              style={Object {}}
            >
              <div
                className="ant-table-content"
              >
                <div
                  className="ant-table-body"
                  onScroll={[Function]}
                  style={Object {}}
                >
                  <table
                    className=""
                    style={Object {}}
                  >
                    <colgroup>
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                    </colgroup>
                    <thead
                      className="ant-table-thead"
                    >
                      <tr
                        style={null}
                      >
                        <th
                          className="ant-table-selection-column"
                        >
                          <span>
                            <div
                              className="ant-table-selection"
                            >
                              <label
                                className="ant-checkbox-wrapper"
                                onMouseEnter={undefined}
                                onMouseLeave={undefined}
                                style={undefined}
                              >
                                <span
                                  className="ant-checkbox ant-checkbox-disabled"
                                  style={Object {}}
                                >
                                  <input
                                    checked={false}
                                    className="ant-checkbox-input"
                                    disabled={true}
                                    name={undefined}
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onClick={undefined}
                                    onFocus={[Function]}
                                    readOnly={undefined}
                                    tabIndex={undefined}
                                    type="checkbox"
                                  />
                                  <span
                                    className="ant-checkbox-inner"
                                  />
                                </span>
                              </label>
                            </div>
                          </span>
                        </th>
                        <th
                          className=""
                        >
                          <span>
                            ID
                          </span>
                        </th>
                        <th
                          className=""
                        >
                          <span>
                            Title
                          </span>
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      className="ant-table-tbody"
                    />
                  </table>
                </div>
                <div
                  className="ant-table-placeholder"
                >
                  <span>
                    <i
                      className="anticon anticon-frown-o"
                    />
                    ææ æ°æ®
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
